### PR TITLE
[FEATURE] Auto-registration of NewContentElementWizard via TCA

### PR DESCRIPTION
**Pull request by linawolf**
Related: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/801
Releases: main

### CODE REVIEW

File: Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst [hunk 1]
```diff
@@ -2,167 +2,197 @@
...
+    of all subpages.
+plugins
+    Plugins provided by extensions
+special
+    Content elements that are used of special cases
 
-   .. code-block:: typoscript
-      :caption: EXT:example/Configuration/page.tsconfig
+All content element groups are listed in
+:php:`$TCA['tt_content']['columns']['CType']['config']['itemGrops']` you can
```
Comment (froemken) [inline Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst:54]:
```suggestion
:php:`$GLOBALS['TCA']['tt_content']['columns']['CType']['config']['itemGrops']` you can
```

File: Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst [hunk 2]
```diff
@@ -2,167 +2,197 @@
...
+debug them in the TYPO3 backend using the backend module
+:guilabel:`System > Configuration` if :composer:`typo3/cms-lowlevel` is installed
+and you are and administrator.
```
Comment (froemken) [inline Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst:57]:
```suggestion
and you are an administrator.
```

File: Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst [hunk 3]
```diff
@@ -2,167 +2,197 @@
...
 
-   .. note::
-      The usage of :file:`Configuration/page.tsconfig` is only valid in TYPO3
-      v12+. If you want to stay compatible with TYPO3 v11 and v12 have a look
-      into :ref:`t3tsconfig:setting-page-tsconfig`.
+Plain content elements or plugins
+=================================
 
-#. :ref:`Register your icon <icon-registration>`
+You can add a plain content element or non-extbase plugin using method
```
Comment (froemken) [inline Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst:67]: In the other documentation a "plain plugin" was the abstract plugin. I'm confused.

```suggestion
You can add a content element or plain plugin (non-extbase) using method
```

File: Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst [hunk 4]
```diff
@@ -2,167 +2,197 @@
...
+Plugins (Extbase) in the "New Content Element" wizard
+=====================================================
 
+To add an Extbase plugin you can use `ExtensionManagementUtility::registerPlugin`
+of class :php:`\TYPO3\CMS\Extbase\Utility\ExtensionManagementUtility`.
 
-Add your plugin or content element to a different tab
-=====================================================
+This method is only available for Extbase plugins defined via
+`ExtensionManagementUtility::addPlugin` in file :file:`EXT:my_extension/ext_localconf.php`
```
Comment (froemken) [inline Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst:104]:
```suggestion
`ExtensionUtility::configurePlugin` in file :file:`EXT:my_extension/ext_localconf.php`
```

File: Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst [hunk 5]
```diff
@@ -2,167 +2,197 @@
...
+Override the wizard with page TSconfig
+======================================
+
+The TCA is always set globally for the complete TYPO3 installation. If you have
+a multi-site installation and want to alter the appearance of content elements
+in the wizard or remove certain content elements this can be done via
+:ref:` page TSconfig <t3tsconfig:setting-page-tsconfig>`.
+This is commonly done on
+
+a per site basis so you can use the :ref:`Site set page TSconfig provider <site-sets-page-tsconfig>`
```
Comment (froemken) [inline Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst:120]: 2 line-breaks here

```suggestion
This is commonly done on a per site basis so you can use the :ref:`Site set page TSconfig provider <site-sets-page-tsconfig>`
```

File: Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst [hunk 6]
```diff
@@ -2,167 +2,197 @@
...
+    :caption: EXT:my_sitepackage/Configuration/Sets/MySet/page.tsconfig
 
-When you look at existing page TSconfig in the :guilabel:`Info` module, you may
-notice that :typoscript:`show` has been set to include all for the
-:guilabel:`Plugins` tab:
+.. _content-element-wizard-create-group:
 
-.. code-block:: typoscript
+Register a new group in the "New Content Element" wizard
+========================================================
```
Comment (froemken) [inline Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst:173]: Note: The tabs/groups are ordered in ASC order by their group keys (not label) by default. With "before" and "after" you can move a group to desired position.

File: Documentation/ApiOverview/ContentElements/_AddingYourOwnContentElements/_page_change_item.tsconfig [hunk 7]
```diff
@@ -0,0 +1,10 @@
+mod.wizards.newContentElement.wizardItems {
+  special.html {
+    title = Plain HTML(use wit care!!)
```
Comment (froemken) [inline Documentation/ApiOverview/ContentElements/_AddingYourOwnContentElements/_page_change_item.tsconfig:3]:
```suggestion
    title = Plain HTML(use with care!!)
```

File: Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst [hunk 8]
```diff
@@ -2,167 +2,197 @@
...
-            'provider' => \TYPO3\CMS\Core\Imaging\IconProvider\SvgIconProvider::class,
-            'source' => 'EXT:example/Resources/Public/Icons/example-registration.svg',
-         ],
-      ];
+When you are using `CType` (recommended) for parameter `$type` the content
+element is added to the select item list of column `CType` in table `tt_content`.
 
-#. After clearing cache, create a new content element
+If you are using the default `list_type` for the parameter it is added as
+subtype.
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst:81]: `list_type` is the field, list-type is English?

File: Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst (hunk 6 above)
Comment (linawolf) [reply Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst:171]: Actually this does not work with pure TCA, see https://typo3.slack.com/archives/C03AM9R17/p1725799806264659

Comment (linawolf) [reply Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst:171]: I would adress this topic in a follow up depending on how the fixes develop

**Review by froemken — APPROVED** [review]

**Pull request merged**